R/read_csv.R

```diff
@@ -52,253 +75,266 @@
+                posterior::as_draws_array(draws[, sampler_diagnostics]),
+                along="chain"
+              )
+            }
         }
-        if (length(sampler_diagnostics) > 0) {
-          new_post_warmup_sampler_diagnostics_draws <- posterior::as_draws_array(draws[(num_warmup_draws+1):all_draws, sampler_diagnostics])
-          if (is.null(post_warmup_sampler_diagnostics_draws)) {
-            post_warmup_sampler_diagnostics_draws <- new_post_warmup_sampler_diagnostics_draws
-          } else {
-            post_warmup_sampler_diagnostics_draws <- posterior::bind_draws(post_warmup_sampler_diagnostics_draws,
-                                                                          new_post_warmup_sampler_diagnostics_draws,
-                                                                          along="chain")
-          }
+      } else if (metadata$method == "variational") {
+        # ignore first line as it's just the mean and lp__ as it's always 0
+        variational_draws <- posterior::as_draws_matrix(
+          draws[-1, colnames(draws) != "lp__", drop=FALSE]
+        )
+        if ("log_p__" %in% posterior::variables(variational_draws)) {
+          variational_draws <- posterior::rename_variables(variational_draws, lp__ = "log_p__")
         }
-      } else {
-        warmup_draws <- NULL
-        warmup_sampler_diagnostics_draws <- NULL
-        if (length(variables) > 0) {
-          new_post_warmup_draws <- posterior::as_draws_array(draws[, variables])
-          if (is.null(post_warmup_draws)) {
-            post_warmup_draws <- new_post_warmup_draws
-          } else {
-            post_warmup_draws <- posterior::bind_draws(post_warmup_draws,
-                                                      new_post_warmup_draws,
-                                                      along="chain")
-          }
-        }
-        if (sampling_info$algorithm != "fixed_param") {
-          if (length(sampler_diagnostics) > 0) {
-            new_post_warmup_sampler_diagnostics_draws <- posterior::as_draws_array(draws[, sampler_diagnostics])
-            if (is.null(post_warmup_sampler_diagnostics_draws)) {
-              post_warmup_sampler_diagnostics_draws <- new_post_warmup_sampler_diagnostics_draws
-            } else {
-              post_warmup_sampler_diagnostics_draws <- posterior::bind_draws(post_warmup_sampler_diagnostics_draws,
-                                                                            new_post_warmup_sampler_diagnostics_draws,
-                                                                            along="chain")
-            }
-          }
+        if ("log_g__" %in% posterior::variables(variational_draws)) {
+          variational_draws <- posterior::rename_variables(variational_draws, lp_approx__ = "log_g__")
         }
+      } else if (metadata$method == "optimize") {
+        point_estimates <- posterior::as_draws_matrix(draws[1,, drop=FALSE])[, variables]
```

Comment:
I added `[, variables]`, otherwise `lp__` never gets dropped even if it's not in the user's specified `variables` argument.